src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const uniqueValues: any[] = [];
+
+  for (const item of items) {
+    const value = item[key];
+
+    if (!uniqueValues.includes(value)) {
+      uniqueValues.push(value);
+    }
+  }
+
+  uniqueValues.forEach((value) => {
```

Comment:
It could be simplified a lot, try to do it without the `uniqueValues` array at all. And actually you can just generate an object with `reduce` method